File: certificates/problem-solving-basic/vowel-substring/findSubstring.c
#include <stdio.h>
#include <string.h>

char* findSubstring(char* s, int k)
{
  char *result = (char *)malloc((k + 1) * sizeof(char));
  char vowels[] = "aeiou";
  int count = 0;
  int ans = 0;
  char *ptr;

  for (int i = 0; i < k; i++)
  {
    ptr = strchr(vowels, s[i]);
    count += (ptr != NULL);
  }
  int max = count;
  for (int i = k; i < strlen(s); i++)
  {
    ptr = strchr(vowels, s[i]);
    count += (ptr != NULL);
    ptr = strchr(vowels, s[i - k]);
    count -= (ptr != NULL);

    if (count > max)
    {
      max = count;
      ans = i - k + 1;
    }
  }

  if (max > 0)
  {
    strncpy(result, s + ans, k);
    result[k] = '\0';
    return result;
  }

  free(result);
  return "Not found!";
}

File: certificates/problem-solving-basic/vowel-substring/findSubstring.js
function findSubstring(s, k) {
  let vowels = ["a", "e", "i", "o", "u"];
  let count = 0;
  let ans = 0;

  for (let i = 0; i < k; i++) {
    count += vowels.includes(s[i]);
  }
  let max = count;

  for (let i = k; i < s.length; i++) {
    count += vowels.includes(s[i]);
    count -= vowels.includes(s[i - k]);
    if (count > max) {
      max = count;
      ans = i - k + 1;
    }
  }
  if (max > 0) {
    return s.slice(ans, ans + k);
  }

  return "Not found!";
}